test/service-worker/testServiceWorker.js

```diff
@@ -0,0 +1,309 @@
+      sinon.stub(self, 'fetch');
+      sinon.stub(oghliner, 'openCache').returns(Promise.resolve(mockedCache));
+      sinon.stub(oghliner, 'extendToIndex').returns('/index.html');
+    });
+
+    afterEach(function () {
+      self.fetch.restore();
+      oghliner.openCache.restore();
+      oghliner.extendToIndex.restore();
+    });
+
+    it('looks for the request first in the cache', function () {
+      var mockedRequest = {};
+      var mockedResponse = {};
+      mockedCache.match.returns(Promise.resolve(mockedResponse));
+      return oghliner.get(mockedRequest).then(function (response) {
+        assert.strictEqual(response, mockedResponse);
+        assert.notOk(self.fetch.called);
+      });
+    });
+
+    it('looks for the request over the network if the cache fails', function () {
+      var mockedRequest = {};
+      var noResponse = undefined;
+      mockedCache.match.returns(Promise.resolve(noResponse));
+      return oghliner.get(mockedRequest).then(function (response) {
+        assert.strictEqual(response, noResponse);
+        assert(self.fetch.calledOnce);
+        assert(self.fetch.calledWith(mockedRequest));
+      });
+    });
+  });
+
+  describe('extendToIndex()', function () {
+    [
+      '/',
+      '/?param=1',
+      '/#section',
+      '/?param=1#section',
+      '/path/',
+      '/path/?param=1',
+      '/path/#section',
+      '/path/?param=1#section',
+    ].forEach(function (path) {
+      var originalUrl = new URL(path, self.location);
+      var originalRequest = new Request(originalUrl);
+
+      it('makes requests to folders to be requests to index.html', function () {
+        var resultRequest = oghliner.extendToIndex(originalRequest);
+        var resultUrl = new URL(resultRequest.url);
+        var folderUrl = new URL(resultUrl);
+        folderUrl.pathname = folderUrl.pathname.replace(/index\.html$/, '');
+
+        // check they are the same except for pathname
+        assert(sameRequestURL(resultUrl, originalUrl, 'pathname'));
+        assert(resultUrl.pathname.endsWith('/index.html'));
+        assert(sameRequestURL(folderUrl, originalUrl));
+      });
+    });
+
+    [
+      '/index.html',
+      '/page.html?param=1',
+      '/page.html#section',
+      '/page.html?param=1#section',
+      '/path/item',
+      '/path/item?param=1',
+      '/path/item#section',
+      '/path/item?param=1#section',
+    ].forEach(function (path) {
+      var originalUrl = new URL(path, self.location);
+      var originalRequest = new Request(originalUrl);
+
+      it('does not alter requests to non folders', function () {
+        var resultRequest = oghliner.extendToIndex(originalRequest);
+        var resultUrl = new URL(resultRequest.url);
+
+        assert(sameRequestURL(resultUrl, originalUrl));
+      });
+    });
+  });
+
+  describe('prepareCache()', function () {
+    function createNonEmptyCache(name) {
+      return self.caches.open(name)
+        .then(function (cache) {
```

Comment:
Nit: for promises, we're using the Q-style in most of the code, e.g.:
```Javascript
  return self.caches.open(name)
  .then(function(cache) {
    // ...
  });
```

It doesn't really matter though.